server/routes/api/v1/users/register.js

```diff
@@ -48,22 +50,94 @@ export default async function (fastify, _opts) {
         inviteId,
       } = request.body;
 
+      const errorList = new Array();
+
+      // Validate request body
+      try {
+        userSchema.parse(request.body);
+      } catch (error) {
+        const parseErrors = JSON.parse(error.message);
+        parseErrors.forEach((e) =>
+          errorList.push({ path: e.path[0], message: e.message }),
+        );
+      }
+
+      // Check email is not duplicated
+      const userFromEmail = await fastify.prisma.user.findUnique({
+        where: { email: email },
```

Comment:
We should make sure that this works regardless of case. I.e. Admin.User@test.com and admin.user@test.com are considered the same.

I think the simplest way to do that might be to just switch the type of the email column in the schema to postgres' "citext" (case-insensitive text) type. Looks like prisma can be configured to include that extension. So from `String` to `@db.Citext`

https://www.prisma.io/docs/orm/prisma-schema/postgresql-extensions#how-to-represent-postgresql-extensions-in-your-prisma-schema

https://www.prisma.io/docs/orm/overview/databases/postgresql#mapping-between-postgresql-database-column-types-to-prisma-orm-scalar-and-native-types

Reply:
Let's modify the test so that it can verify it works regardless of case...

Reply:
Citext extension added!